proto/instrument.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package rs.tokio.console.instrument;
+
+import "google/protobuf/timestamp/timestamp.proto";
+import "common.proto";
+import "tasks.proto";
+import "resources.proto";
+import "async_ops.proto";
+import "resource_ops.proto";
+
+service Instrument {
+    rpc WatchUpdates(InstrumentRequest) returns (stream InstrumentUpdate) {}
+    rpc WatchTaskDetails(TaskDetailsRequest) returns (stream tasks.TaskDetails) {}
+}
+
+message  InstrumentRequest{
+}
+
+message TaskDetailsRequest {
+    common.SpanId id = 1;
+}
+
+message InstrumentUpdate {
+    // The system time when this update was recorded.
+    //
+    // This is the timestamp any durations in the included `Stats` were
+    // calculated relative to.
+    google.protobuf.Timestamp now = 1;
+
+    // Task state update.
+    tasks.TaskUpdate task_update = 2;
+
+    // Resource state update.
+    resources.ResourceUpdate resource_update = 3;
+
+    // Async operations state update
+    async_ops.AsyncOpUpdate async_op_update = 4;
+
+    // Resource operations state update
+    resource_ops.ResourceOpUpdate resource_op_update = 5;
```

Comment:
hmmm...at first glance, it seems kind of weird to me to combine the tasks, resources, and async ops into one big RPC; i'd kind of naively expect to be able to watch them separately.

but, after giving it some thought, I can understand the motivation for combining them: the async ops reference the IDs of both resource spans and task spans, so that data may not make as much sense on its own. also, this allows anchoring all the data with one shared timestamp, which seems good (as well as sharing the `RegisterMetadata` updates so we don't have to send a bunch of separate ones).

if we're doing it this way, and i think this might be the best way to do it, it would be nice to add a comment to `InstrumentUpdate` explaining why all this data is combined?

Reply:
also, it might make sense in the future for `InstrumentRequest` to allow selecting which updates we want to recieve. that would allow applications to pick and choose what data they're interested in, but leaving everything combined in one update message so that when we are watching multiple kinds of data we can easily cross-reference things and not have to manage different timestamps. but that's definitely something to work on in a future change, not now.